src/components/Login/Login.jsx

```diff
@@ -36,16 +37,15 @@ const Login = ({ changeLog }) => {
 			password,
 		};
 
-		const res = await fetchLogin(user, history);
-		if (res) {
-			dispatch(actionCreators.login(res));
-		}
-		await localStorage.setItem('token', res.result);
-		if (localStorage.getItem('token') !== 'undefined') {
+		await dispatch(getUser(user));
+	};
+
+	useEffect(() => {
+		if (token) {
 			history.push('/courses');
+			changeLog(true);
```

Comment:
you don't seem to need `changeLog` state anymore: all your logged-in checks can be performing using store token check